src/factor.jl

```diff
@@ -58,6 +58,7 @@ function factor_graph(
         sp = spectrum(cl, num_states=get(num_states_cl, v, basis_size(cl)))
         set_prop!(fg, v, :spectrum, sp)
         set_prop!(fg, v, :loc_en, vec(sp.energies))
+        set_prop!(fg, v, :loc_dim, length(vec(sp.energies)))
```

Comment:
This property is redundant. Furthermore it might impact data coherence, as `loc_en`, `spectrum` and `loc_dim` can in principle be all changed independently.

Reply:
I agree. That said, I do not really want to mess with this right now. This will change anyway so I would suggest leaving it for the time being.